src/rust/include/rust/orchard/wallet.h

```diff
@@ -287,6 +291,29 @@ OrchardSpendInfoPtr* orchard_wallet_get_spend_info(
         const unsigned char txid[32],
         uint32_t action_idx);
 
+/**
+ * Run the garbage collection operation on the wallet's note commitment
+ * tree.
+ */
+bool orchard_wallet_gc_note_commitment_tree(OrchardWalletPtr* wallet);
+
+/**
+ * Write the wallet's note commitment tree to the provided stream.
+ */
+bool orchard_wallet_write_note_commitment_tree(
+        const OrchardWalletPtr* wallet,
+        void* stream,
+        write_callback_t write_cb);
+
+/**
+ * Read a note commitment tree from the provided stream, and update the wallet's internal
+ * note commitment tree state to equal the value that was read.
```

Comment:
I presume this returns `false` on I/O error?

Reply:
Either on IO error, or on a violation of the invariants of tree construction.